### PR TITLE
Local containers

### PR DESCRIPTION
**Pull request by mpeels**
## Newly added
1. Updates the docker compose to use the existing `dataingestion.env`
2. Updates the README with newly required fields
initial
keycloak

## Notes - Previously reviewed

1. Adds a docker container for the NBS databases.
4. Adds a docker container for a local keycloak instance to simplify local development
5. Adds support for an `application-local.yaml` for the data-ingestion-service.
6. Updates gradle from `8.0` to `8.10.1`

## Checklist

- [ ] PR focuses on a single story.
- [ ] New unit tests added and ensured they pass.
- [x] Service has been tested in local and it works as expected.
- [x] Documentation has been updated for this code change (if needed).
- [x] Code follows the Java Coding Conventions (https://www.oracle.com/java/technologies/javase/codeconventions-programmingpractices.html).

## Types of changes

What types of changes does this PR introduces?

- [ ] Bugfix
- [ ] New feature
- [ ] Breaking change
- [x] Local dev improvements

## Testing

- [ ] Does this PR has >90% code coverage?
- [ ] Is the screenshot attached for code coverage?
- [x] Does the `gradle build` pass in your local? 
- [x] Is the `gradle build` logs attached?

image

### CODE REVIEW

File: docker-compose.yml
Comment (ssathiah) [inline docker-compose.yml:1]: Does dataingestion.env require any new variables?

**Review by ssathiah — APPROVED** [review]
Looks good.

File: docker-compose.yml
Comment (mpeels) [reply docker-compose.yml:1]: By default the docker-compose.yml will read any `.env` file. So currently I have the `DATABASE_PASSWORD` and `KEYCLOAK_ADMIN_PASSWORD` specified in it.

Comment (mpeels) [reply docker-compose.yml:1]: I don't like having two `.env` files here. I'll consolidate my new `.env` into the existing `dataingestion.env` and update the README

Comment (ssathiah) [reply docker-compose.yml:1]: Thanks.

**Pull request merged**